packages/astro/src/core/build/plugins/plugin-prerender.ts

```diff
@@ -1,14 +1,25 @@
 import type { Rollup, Plugin as VitePlugin } from 'vite';
 import type { BuildInternals } from '../internal.js';
-import type { AstroBuildPlugin } from '../plugin.js';
 import type { StaticBuildOptions } from '../types.js';
 import { ASTRO_PAGE_RESOLVED_MODULE_ID } from './plugin-pages.js';
 import { getPagesFromVirtualModulePageName } from './util.js';
 
-function vitePluginPrerender(internals: BuildInternals): VitePlugin {
+export function pluginPrerender(
+	opts: StaticBuildOptions,
+	internals: BuildInternals,
+): VitePlugin | undefined {
+	// Static output can skip prerender completely because we're already rendering all pages
+	if (opts.settings.buildOutput === 'static') {
```

Comment:
I don't think this is true, we always want to create the prerender output.

Reply:
It's possible. However, consider that our build is still broken due to `createRoutesList` being called twice, which creates many false positives and/or many false negatives. 

Also, if we still need this check in the future, I would prefer this being done by the caller. The current logic isn't very intuitive. A function plugin shouldn't decide to return a plugin or not, the caller should decide